app/v2/connectors/PersistenceConnector.scala

```diff
@@ -216,4 +228,23 @@ class PersistenceConnectorImpl @Inject() (httpClientV2: HttpClientV2, appConfig:
       .executeAndExpect(OK)
   }
 
+  def updateMessageBody(
+    eoriNumber: EORINumber,
+    movementType: MovementType,
+    movementId: MovementId,
+    messageId: MessageId,
+    source: Source[ByteString, _]
+  )(implicit
+    hc: HeaderCarrier,
+    ec: ExecutionContext
+  ): Future[Unit] = {
+    val url = appConfig.movementsUrl.withPath(postMessageBodyUrl(eoriNumber, movementType, movementId, messageId))
+
+    httpClientV2
+      .post(url"$url")
```

Comment:
Here we need to pass the message Type as part of header. Just look on how it is passed for other endpoint in this class.Add X-Message-Type with header

Reply:
Done